src/binding/python/Attributable.cpp

```diff
@@ -49,14 +51,151 @@ using namespace openPMD;
 using PyAttributeKeys = std::vector< std::string >;
 //PYBIND11_MAKE_OPAQUE(PyAttributeKeys)
 
+void setAttributeFromBufferInfo(
+    Attributable & attr,
+    std::string const& key,
+    py::buffer_info const& buf
+) {
+    // Numpy: Handling of arrays and scalars
+    // work-around for https://github.com/pybind/pybind11/issues/1224
+    // -> passing numpy scalars as buffers needs numpy 1.15+
+    //    https://github.com/numpy/numpy/issues/10265
+    //    https://github.com/pybind/pybind11/issues/1224#issuecomment-354357392
+    // scalars, see PEP 3118
+    // requires Numpy 1.15+
+    if( buf.ndim == 0 ) {
+        // refs:
+        //   https://docs.scipy.org/doc/numpy-1.15.0/reference/arrays.interface.html
+        //   https://docs.python.org/3/library/struct.html#format-characters
+        // std::cout << "  scalar type '" << buf.format << "'" << std::endl;
+        // typestring: encoding + type + number of bytes
+        if( buf.format.find("h") != std::string::npos )
+            attr.setAttribute( key, *const_cast<short*>(static_cast<short*>(buf.ptr)) );
+        else if( buf.format.find("i") != std::string::npos )
+            attr.setAttribute( key, *const_cast<int*>(static_cast<int*>(buf.ptr)) );
+        else if( buf.format.find("l") != std::string::npos )
+            attr.setAttribute( key, *const_cast<long*>(static_cast<long*>(buf.ptr)) );
```

Comment:
@C0nsultant hm, somehow this is still `const` xD Am I missing something?
https://travis-ci.org/openPMD/openPMD-api/jobs/424882144

Reply:
Huh, or I really need to write `Attribute(*const_cast<T*>(&value))` in the C++ backend...

Reply:
e.g. on clang OSX in Travis, `long` is `long long` but only one of them is in the variant list.
On MSVC, `long` and `int` are identical, etc.